lib/auth/auth_with_roles.go

```diff
@@ -4591,6 +4591,11 @@ func (a *ServerWithRoles) UpsertTrustedCluster(ctx context.Context, tc types.Tru
 }
 
 func (a *ServerWithRoles) ValidateTrustedCluster(ctx context.Context, validateRequest *ValidateTrustedClusterRequest) (*ValidateTrustedClusterResponse, error) {
+	// Don't allow leaf clusters if running in Cloud.
+	if modules.GetModules().Features().Cloud {
+		return nil, trace.NotImplemented("cloud clusters do not support trusted cluster resources")
```

Comment:
Maybe, so we can capitalize Teleport Cloud:

```suggestion
		return nil, trace.NotImplemented("trusted cluster resources not supported by Teleport Cloud")
```